### PR TITLE
Add Builder for ClassifierCompositeItemWriter

### PR DESCRIPTION
**Pull request by cppwfs**
resolves BATCH-2599

Added null check for setClassifier in ClassifierCompositeItemWriter.

### CODE REVIEW

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	/**
+	 * Establish the classifier to be used for the selection of which {@link ItemWriter}
+	 * to use.
+	 *
+	 * @param classifier the classifier to set
+	 * @return this instance for method chaining
+	 * @see org.springframework.batch.item.support.ClassifierCompositeItemWriter#setClassifier(Classifier)
+	 */
+	public ClassifierCompositeItemWriterBuilder classifier(Classifier<T, ItemWriter<? super T>> classifier) {
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java]: Missing generic on return

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Creates a fully qualified ClassifierCompositeItemWriter.
+ *
+ * @author Glenn Renfro
+ *
+ * @since 4.0
+ */
+public class ClassifierCompositeItemWriterBuilder<T> {
+
+	private Classifier<T, ItemWriter<? super T>> classifier = new ClassifierSupport<T, ItemWriter<? super T>>(null);
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java]: Why are we providing a non-valid default?

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	/**
+	 * Establish the classifier to be used for the selection of which {@link ItemWriter}
+	 * to use.
+	 *
+	 * @param classifier the classifier to set
+	 * @return this instance for method chaining
+	 * @see org.springframework.batch.item.support.ClassifierCompositeItemWriter#setClassifier(Classifier)
+	 */
+	public ClassifierCompositeItemWriterBuilder<T> classifier(Classifier<T, ItemWriter<? super T>> classifier) {
+		Assert.notNull(classifier, "A classifier is required.");
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java]: Move this assert into the build.  As it sits right now, you can end up with an invalid writer (one with the invalid default) by just doing:
```
new ClassifierCompositeItemWriter().build();
```
A test case for this condition should also be added.

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java (hunk 2 above)
Comment (cppwfs) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java]: Good point

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java (hunk 3 above)
Comment (cppwfs) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/support/builder/ClassifierCompositeItemWriterBuilder.java]: done

**Comment by mminella** [comment]
LGTM.  Squashed and merged as 70a1a83a9a7a43fdd7e2a3a199f7c989a825d18a

**Pull request closed without merging**